deployments/kubernetes/install/scripts/install-cni.sh

```diff
@@ -32,6 +32,25 @@ MOUNTED_CNI_NET_DIR=${MOUNTED_CNI_NET_DIR:-/host/etc/cni/net.d}
 CNI_CONF_NAME=${CNI_CONF_NAME:-$(ls ${MOUNTED_CNI_NET_DIR} | head -n 1)}
 CNI_CONF_NAME=${CNI_CONF_NAME:-10-calico.conflist}
 KUBECFG_FILE_NAME=${KUBECFG_FILE_NAME:-ZZZ-istio-cni-kubeconfig}
+CFGCHECK_INTERVAL=${CFGCHECK_INTERVAL:-1}
+
+function check_install() {
+  cfgfile_nm=$(ls ${MOUNTED_CNI_NET_DIR} | head -n 1)
+  if [[ "${cfgfile_nm}" != "${CNI_CONF_NAME}" ]]; then
+    echo "ERROR: CNI config file \"${MOUNTED_CNI_NET_DIR}/${CNI_CONF_NAME}\" preempted by \"$cfgfile_nm\"."
+    exit 1  
+  fi
+  if [ -e "${MOUNTED_CNI_NET_DIR}/${CNI_CONF_NAME}" ]; then
+    istiocni_conf=$(cat ${MOUNTED_CNI_NET_DIR}/${CNI_CONF_NAME} | jq '.plugins[]? | select(.type == "istio-cni")')
+    if [[ "$istiocni_conf" == "" ]]; then
+      echo "ERROR: istio-cni CNI config removed from file: \"${MOUNTED_CNI_NET_DIR}/${CNI_CONF_NAME}\""
+      exit 1  
+    fi
+  else
+    echo "ERROR: CNI config file \"${MOUNTED_CNI_NET_DIR}/${CNI_CONF_NAME}\" removed."
+    exit 1  
```

Comment:
Hi Venil,   It will loop forever if nothing changes just like it does today.  But if any of these checks in `check_install` hit then it will exit the proc and k8s will restart the pod.  When the pod restarts, the install-cni.sh config apply steps of the script will rerun and the istio-cni config will be applied again and go back into the check_install() loop.

So, you will be able to tell when something messed up the istio-cni config by how many times the `istio-cni-node` pods restart on a node.

Reply:
Thanks for the clarification.